### PR TITLE
added create resource account and publish package method with a test …

### PR DESCRIPTION
**Pull request by dhruvja**
### Description

This PR is regarding the issue: https://github.com/aptos-labs/aptos-core/issues/5349

### CODE REVIEW

File: ecosystem/typescript/sdk/src/tests/e2e/aptos_client.test.ts [hunk 1]
```diff
@@ -533,6 +533,43 @@ test(
...
+          ).toUint8Array(),
+        ),
+      ],
+    );
+
+    await client.waitForTransaction(txnHash);
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/tests/e2e/aptos_client.test.ts:565]:
```suggestion
    await client.waitForTransaction(txnHash,{ checkSuccess: true });
```

and then we can remove lines 657-568

File: ecosystem/typescript/sdk/src/providers/aptos_client.ts [hunk 2]
```diff
@@ -770,6 +770,41 @@ export class AptosClient {
...
+   * `packageMetadata` and `modules` can be generated with command
+   * `aptos move compile --save-metadata [ --included-artifacts=<...> ]`. 
+   * @param sender
+   * @param seed seeds for creation of resource address
+   * @param packageMetadata package metadata bytes
+   * @param modules bytecodes of modules
+   * @param extraArgs
+   */
+
+  async createResourceAccountAndpublishPackage(
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/providers/aptos_client.ts:784]:
```suggestion
  async createResourceAccountAndPublishPackage(
```

File: ecosystem/typescript/sdk/src/providers/aptos_client.ts [hunk 3]
```diff
@@ -770,6 +770,41 @@ export class AptosClient {
...
 
+  /**
+   * Publishes a move packages by creating a resource account. The package cannot be upgraded since it is deployed by resource account
+   * `packageMetadata` and `modules` can be generated with command
+   * `aptos move compile --save-metadata [ --included-artifacts=<...> ]`. 
+   * @param sender
+   * @param seed seeds for creation of resource address
+   * @param packageMetadata package metadata bytes
+   * @param modules bytecodes of modules
+   * @param extraArgs
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/providers/aptos_client.ts:781]:
```suggestion
   * @param extraArgs
   * @returns Transaction hash
```

**Review by 0xmaayan — COMMENTED** [review]
Thank you for looking into it! 
I wonder if it makes more sense to introduce a `ResourceAccount` class with this function, and then we would have this class holds all the `resource_account.move` module [entry functions](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/resource_account.move#L84) and the SDK [getResourceAccountAddress](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/account/aptos_account.ts#L115) generate method

**Comment by dhruvja** [comment]
> Thank you for looking into it! I wonder if it makes more sense to introduce a `ResourceAccount` class with this function, and then we would have this class holds all the `resource_account.move` module [entry functions](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/resource_account.move#L84) and the SDK [getResourceAccountAddress](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/account/aptos_account.ts#L115) generate method

Well not a bad idea, but there are hardly any functions entry functions in `resource_account.move` atm.

**Comment by 0xmaayan** [comment]
@dhruvja Please fix `Lint+Test` test, than I will approve

**Review by davidiw — APPROVED** [review]

File: ecosystem/typescript/sdk/src/providers/aptos_client.ts [hunk 4]
```diff
@@ -770,6 +770,43 @@ export class AptosClient {
...
+   * @returns Transaction hash
+   */
+
```
Comment (davidiw) [inline ecosystem/typescript/sdk/src/providers/aptos_client.ts:785]:
```suggestion
```

**Review by 0xmaayan — APPROVED** [review]

**Pull request merged**